src/DataStructures/Tensor/EagerMath/OrthonormalOneform.cpp

```diff
@@ -0,0 +1,126 @@
+    const Scalar<DataType>& det_spatial_metric) noexcept {
+  *orthonormal_form = cross_product(first_unit_form, second_unit_form,
+                                    spatial_metric, det_spatial_metric);
+}
+
+template <typename DataType, typename Frame>
+tnsr::i<DataType, 3, Frame> orthonormal_oneform(
+    const tnsr::i<DataType, 3, Frame>& first_unit_form,
+    const tnsr::i<DataType, 3, Frame>& second_unit_form,
+    const tnsr::ii<DataType, 3, Frame>& spatial_metric,
+    const Scalar<DataType>& det_spatial_metric) noexcept {
+  tnsr::i<DataType, 3, Frame> orthonormal_form{};
+  orthonormal_oneform(make_not_null(&orthonormal_form), first_unit_form,
+                      second_unit_form, spatial_metric, det_spatial_metric);
+  return orthonormal_form;
+}
+
+#define DTYPE(data) BOOST_PP_TUPLE_ELEM(0, data)
+#define FRAME(data) BOOST_PP_TUPLE_ELEM(1, data)
+#define DIM(data) BOOST_PP_TUPLE_ELEM(2, data)
+
+#define INSTANTIATE_FIRST_FORM(_, data)                                      \
+  template void orthonormal_oneform(                                         \
+      const gsl::not_null<tnsr::i<DTYPE(data), DIM(data), FRAME(data)>*>     \
+          orthonormal_form,                                                  \
+      const tnsr::i<DTYPE(data), DIM(data), FRAME(data)>& unit_form,         \
+      const tnsr::II<DTYPE(data), DIM(data), FRAME(data)>&                   \
+          inv_spatial_metric) noexcept;                                      \
+  template tnsr::i<DTYPE(data), DIM(data), FRAME(data)> orthonormal_oneform( \
+      const tnsr::i<DTYPE(data), DIM(data), FRAME(data)>& unit_form,         \
+      const tnsr::II<DTYPE(data), DIM(data), FRAME(data)>&                   \
+          inv_spatial_metric) noexcept;
+
+GENERATE_INSTANTIATIONS(INSTANTIATE_FIRST_FORM, (double, DataVector),
+                        (Frame::Inertial), (2, 3))
+
+#define INSTANTIATE_SECOND_FORM(_, data)                             \
+  template void orthonormal_oneform(                                 \
+      const gsl::not_null<tnsr::i<DTYPE(data), 3, FRAME(data)>*>     \
+          orthonormal_form,                                          \
+      const tnsr::i<DTYPE(data), 3, FRAME(data)>& first_unit_form,   \
+      const tnsr::i<DTYPE(data), 3, FRAME(data)>& second_unit_form,  \
+      const tnsr::ii<DTYPE(data), 3, FRAME(data)>& spatial_metric,   \
+      const Scalar<DTYPE(data)>& det_spatial_metric) noexcept;       \
+  template tnsr::i<DTYPE(data), 3, FRAME(data)> orthonormal_oneform( \
+      const tnsr::i<DTYPE(data), 3, FRAME(data)>& first_unit_form,   \
+      const tnsr::i<DTYPE(data), 3, FRAME(data)>& second_unit_form,  \
+      const tnsr::ii<DTYPE(data), 3, FRAME(data)>& spatial_metric,   \
+      const Scalar<DTYPE(data)>& det_spatial_metric) noexcept;
+
+GENERATE_INSTANTIATIONS(INSTANTIATE_SECOND_FORM, (double, DataVector),
+                        (Frame::Inertial))
+
+#undef DTYPE
+#undef DIM
+#undef FRAME
```

Comment:
[super nitpick, non binding:] Switch DIM, FRAME here so order of undefs matches order of defs above

Reply:
actually you should `#undef` in the opposite order of the `#define`s

Reply:
I don't believe we do this consistently in the codebase. I suggest @carmaza leave this as is and I'll put in a cleanup commit shortly to address this?